substrate/frame/revive/rpc/src/client/runtime_api.rs

```diff
@@ -65,12 +69,28 @@ impl RuntimeApi {
 	pub async fn dry_run(
 		&self,
 		tx: GenericTransaction,
+		block: BlockNumberOrTagOrHash,
 	) -> Result<EthTransactInfo<Balance>, ClientError> {
-		let payload = subxt_client::apis().revive_api().eth_transact(tx.into());
+		let timestamp_override = match block {
+			BlockNumberOrTagOrHash::BlockTag(BlockTag::Pending) => {
+				Some(Timestamp::current().as_millis())
+			}
+			_ => None,
+		};
+		let payload = subxt_client::apis().revive_api().eth_transact(tx.into(), timestamp_override);
 		let result = self.0.call(payload).await?;
 		match result {
 			Err(err) => {
-				log::debug!(target: LOG_TARGET, "Dry run failed {err:?}");
+				// Attempt to decode revert reason if available
+				let decoded_msg = match &err.0 {
+					EthTransactError::Data(data) => match decode_revert_reason(data) {
+						Some(reason) => format!("execution reverted: {reason}"),
+						None => "execution reverted".to_string(),
+					},
+					EthTransactError::Message(msg) => msg.clone(),
+				};
+				log::debug!(target: LOG_TARGET, "err = {:?}", err);
+				log::debug!(target: LOG_TARGET, "Dry run failed {decoded_msg}");
```

Comment:
let's remove these logs this can be done by cast directl rather than doing logging on the server